compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -514,6 +514,8 @@ fn inner_parse_loop<'root, 'tt>(
 
         // If `idx >= len`, then we are at or past the end of the matcher of `item`.
         if idx >= len {
+            debug_assert!(matches!(item.top_elts, Tt(TokenTree::Sequence(..))));
```

Comment:
Since this is the assert triggering on `libcore`, I have locally changed this to `debug_assert_matches!` (note: it's not in the prelude and needs to be imported) and saw that `top_elts` is a `TtSeq` of two `TokenTree::Sequence`s in this case.

Reply:
Thanks for the info. I ran the whole test suite locally, not sure why I didn't get this failure. I'll just take out the commit that added the assertion.

Reply:
Oh, it was a `debug_assert` and I was not running a debug build. I fixed the problem by moving the assertion within the subsequent `if`, which was where I originally put it but then I got overconfident and moved it outwards :)